NL02 ingestion

yasmin-evans opened the pull request:
NL02 ingest without BC18 commits

* Additional cleanup commit for remaining data_temp files

yasmin-evans committed:
Added NL02 translation and conversion scripts from master CASFRI branch to testing branch

* load_NL02_*.sh scripts added

* NL02 attributes added to attribute dependencies table

* NL02 ingestion branch created

yasmin-evans committed:
Typo fixed in load_Nl02 for DROP TABLE StockNonCom

yasmin-evans committed:
Additional translation tables added from CASFRI main brnahc for testing

yasmin-evans committed:
NL02 specific helper functions added to helperFunctionsCASFRI.sql

yasmin-evans committed:
01 hdr table updated to contain NL02 inventory

yasmin-evans committed:
Addition of 02_NL02.sql for translation

yasmin-evans committed:
Updates made to helperFunctionsCASFRI for NL02 Ingestion

yasmin-evans committed:
Further updates to the HelperFunctionsCASFRI script for NL02

* Added missing isForest function and error codes

yasmin-evans committed:
Addition to helperFunctionsCasfri and change to nl_load_photoyear

* added TT_nl_nli01_productivity_type_translation to helperFunctionCasfri

* changed year to photoyear AS year in nl_load_photoyear

yasmin-evans committed:
Change made from photoyear to photo_year in load_nl02_photoyear

yasmin-evans committed:
Added tt_nl_nli02_wetland_code to the helperfunctions creation script

yasmin-evans committed:
Added T_nl_nli02_origin_newfoundland_validation and tt_nl_nli02_wetland_translation to helperfunctions

yasmin-evans committed:
Added T_nl_nli02_origin_lower_validation to the helperFunctions script

yasmin-evans committed:
Test translation tables scripts altered to run for the nl02 ingestion

yasmin-evans committed:
Adding data_temp folder to testing branch for translation testing NL02

yasmin-evans committed:
Added NL02 rawfri ingestion count to checkCounts.sql

Commit:
Changes to test_dst.sql such that NL02 test table is created

yasmin-evans committed:
Changes to translation test file creation scripts such that NL02 is generated correctly

* Added necessary TT_Prepare statements

* Changed TT_translate functions to match NL02.sql calls

Commit:
NL02 test tables dumped, moving to local repo to alter test tables in data13 before comparing tests

yasmin-evans committed:
Edits made to test_cas script to generate the NL02 test table

Commit:
Tables for *_all_new were stale, re-ran dump to generate tests correctly, also applied fixes to select .sql test files such that they generate the correct test tables or NL02

yasmin-evans committed:
Updated data13 .csv files with NL02 test translation rows

Commit:
Test table code cleanup for NL02

yasmin-evans committed:
NL02 added to inventory_list_cas05.csv

Commit:
Update to remove data_temp files

* Remove data_temp file and contents

* Update PG versions to 13 from temp

mtCarto commented on translation/test/test_eco.sql:
```diff
@@ -151,7 +153,7 @@ INSERT INTO casfri50_test.eco_all_new
 ------------------------
 SELECT TT_CreateMappingView('rawfri', 'sk01', 'sk_utm', 700, NULL, 'eco'); -- Generates 0 ECO rows
-INSERT INTO casfri50_test.eco_all_new 
+INSERT INTO casftype filter textri50_test.eco_all_new 
```
this looks like a typo?

mtCarto commented on translation/test/test_eco.sql:
```diff
@@ -191,7 +193,7 @@ INSERT INTO casfri50_test.eco_all_new
 ------------------------
 SELECT TT_CreateMappingView('rawfri', 'ns02', 'ns_nsi', 3000, NULL, 'eco'); -- Generates 254 ECO rows
-INSERT INTO casfri50_test.eco_all_new 
+type filter textINSERT INTO casfri50_test.eco_all_new 
```
another typo

mtCarto commented on translation/test/test_eco.sql:
```diff
@@ -229,9 +231,13 @@ SELECT * FROM TT_Translate_mb_fli_eco_test('rawfri', 'mb07_l1_to_mb_fli_l1_map_1
 SELECT * FROM TT_Translate_nl_nli_eco_test('rawfri', 'nl01_l1_to_nl_nli_l1_map_900_eco');
-------------------------
+------------------------ 
+SELECT TT_CreateMappingView('rawfri', 'nl02', 'nl_nli2', 900, NULL, 'eco'); -- Generates 0 ECO rows
```
does this actually generate 0 rows? if so increase the number so we get some rows, otherwise update the comment

mtCarto commented on translation/test/test_eco.sql:
```diff
@@ -229,9 +231,13 @@ SELECT * FROM TT_Translate_mb_fli_eco_test('rawfri', 'mb07_l1_to_mb_fli_l1_map_1
+------------------------ 
 SELECT TT_CreateMappingView('rawfri', 'qc01', 'qc_ini03', 10000, NULL, 'eco'); -- Generates about 1000 (1051) ECO rows
-INSERT INTO casfri50_test.eco_all_new 
+INSERT INTO casfri50_tesOt.eco_all_new 
```
typo

mtCarto commented on workflow/02_produceCASFRI/01_hdr.sql:
```diff
@@ -43,14 +43,14 @@ SELECT inventory_id, jurisdiction, owner_type, owner_name, standard_type, standa
-  'SK01', 'SK02', 'SK03', 'SK04', 'SK05', 'SK06',
-  'YT01', 'YT02', 'YT03', 'YT04');
+WHERE inventory_id IN ('AB03', 'AB06', 'AB07', 'AB08', 'AB10', 'AB11', 'AB16', 'AB25', 'AB27','AB29', 'AB30', 
+                       'AB31', 'AB32', 'AB33', 'BC08', 'BC10', 'BC11', 'BC12','BC13', 'BC14', 'BC16', 'BC18',
```
are these new tabs? remove if so to make diff easier to read

mtCarto reviewed (changes requested):
Some fixes required in a few of the SQL files. Also need to add NL02 to rawfri check counts in conversion/sh

yasmin-evans committed:
Update helperFunctionsCASFRI.sql

Removed additional tt_nl_nli02_wetland_code

yasmin-evans committed:
Update test_eco.sql

INSERT INTO casfri50_test.eco_all_new replaced with Insert Into casfri50_test.eco_all_new

yasmin-evans committed:
Update 01_hdr.sql

Tabs removed from inventory_id

yasmin-evans committed:
Update test_eco.sql

Typo in INSERT INTO casfri50_tesOt.eco_all_new fixed

yasmin-evans committed:
Update test_eco.sql

NL02 Eco generates 92 rows not 0

yasmin-evans committed:
Update test_eco.sql

type filter textINSERT INTO casfri50_test.eco_all_new typo fixed

yasmin-evans committed:
Update load_test_tables.sh

pg in this file should be 12 not 11

yasmin-evans committed:
Update 01_hdr.sql

Added single space for readability and continuity with previous file format.

yasmin-evans reviewed (commented):
Completed reviewing code change suggestions.

mtCarto reviewed (approved):
Looks good now :+1:
